homeassistant/components/withings/common.py

```diff
@@ -497,23 +550,42 @@ def webhook_config(self) -> WebhookConfig:
     def _do_interval(
         self, action: Callable[[], Coroutine], delay: timedelta, interval: timedelta
     ) -> Callable:
+        """Run action now and schedule it run in an interval for future calls.
+
+        This method has the same functionality of async_track_time_interval except this
+        method also runs the action immediately (with slight delay).
+        :param action: The async function to call.
+        :param delay: Delay the first run by some time. This is good for adding jitter
+        to API calls.
+        :param interval: How much time should elapse between repeats calls of "action".
+        :return: A function for cancelling all future calls.
+        """
         cancel_func1 = None
         cancel_func2 = None
 
         @callback
-        def runner(now: datetime.datetime) -> None:
-            self._hass.async_run_job(action())
+        async def runner(now: datetime.datetime) -> None:
+            await self._hass.async_add_job(action)
```

Comment:
If you know that you're always passing in async functions, you could just do this:
```suggestion
            await action()
```